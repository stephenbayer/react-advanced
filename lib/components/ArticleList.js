import React from 'react';
import Article from './Article';

export default (props) => {
  return (
    <div className={'articlelist'}>
      {Object.values(props.articles).map(article =>
          <Article
            key={article.id}
            article={article}
            actions={props.actions}
          />
      )}
    </div>
  );
};