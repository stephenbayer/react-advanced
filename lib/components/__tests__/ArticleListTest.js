import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' }
    },
    actions: {
      lookupAuthor: (authorId) => ({ id: authorId})
    }
  };
  it('renders correctly', () => {
    const tree = renderer.create(
      <ArticleList
        {...testProps}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

    expect(tree.children.length === 2);

  });
});
