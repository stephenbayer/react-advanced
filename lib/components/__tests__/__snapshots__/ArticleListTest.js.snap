// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div
  className="articlelist"
>
  <div>
    <h2 />
    <div />
    <h3>
      <a>
         
      </a>
    </h3>
    <p />
  </div>
  <div>
    <h2 />
    <div />
    <h3>
      <a>
         
      </a>
    </h3>
    <p />
  </div>
</div>
`;
